### PR TITLE
chore: extract init to reth-db-common #8171

### LINKED ISSUE #8171: extract node-core/init.rs from node-core
### Describe the feature

these operations are out of place here:

https://github.com/paradigmxyz/reth/blob/aa07257d3ba071188b7b7e372856b50be094e8ba/crates/node-core/src/init.rs

they also can't be included in any existing storage crates, like db/provider and should be moved to `reth-db-{common,core,init,ops}`. Any other naming ideas @Rjected ?
leaning towards reth-db-common

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by willco-1**
extracts init to reth-db-common. Let me know if there is any review notes!

closes #8171

### CODE REVIEW

**Review by emhane — COMMENTED** [review]
I can't see the issue unfortunately, think it requires specific credentials

File: crates/reth-db-common/src/lib.rs
Comment (mattsse) [inline crates/reth-db-common/src/lib.rs:1]: this needs the regular lib.rs header, see other crates

also include the `unused` lint and remove unused deps

**Review by mattsse — CHANGES_REQUESTED** [review]
this is great.

only stylistic nits

**Comment by emhane** [comment]
need any help here with merge conflicts @willco-1 ?

**Comment by emhane** [comment]
my bad, closing in favour of https://github.com/paradigmxyz/reth/pull/8373

**Pull request closed without merging**